server/src/main/java/org/opensearch/indices/cluster/IndicesClusterStateService.java

```diff
@@ -203,13 +204,23 @@ public IndicesClusterStateService(
     ) {
         this.settings = settings;
         this.checkpointPublisher = checkpointPublisher;
-        this.buildInIndexListener = Arrays.asList(
-            peerRecoverySourceService,
-            recoveryTargetService,
-            segmentReplicationTargetService,
-            searchService,
-            snapshotShardsService
-        );
+        // if segrep feature flag is not enabled, don't wire the target serivce as an IndexEventListener.
+        if (FeatureFlags.isEnabled(FeatureFlags.REPLICATION_TYPE)) {
+            this.buildInIndexListener = Arrays.asList(
+                peerRecoverySourceService,
+                recoveryTargetService,
+                searchService,
+                snapshotShardsService
+            );
+        } else {
+            this.buildInIndexListener = Arrays.asList(
+                peerRecoverySourceService,
+                recoveryTargetService,
+                segmentReplicationTargetService,
+                searchService,
+                snapshotShardsService
+            );
+        }
```

Comment:
Can we try below simplication ?
```
if (FeatureFlags.isEnabled(FeatureFlags.REPLICATION_TYPE)) {
   this.buildInIndexListener.add(segmentReplicationTargetService);
}
```

Reply:
Arrays.asList returns a mutable list, but it is fixed-size and will throw if we try to add an element.  I wanted to preserve this property & was being quick here.  I've updated to create it as mutable and then make immutable.